### PR TITLE
Fixes #22804 - Parse origin for failed runs

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
Needs https://github.com/ansible/ansible/pull/55437

File: app/services/foreman_ansible/ansible_report_scanner.rb [hunk 1]
```diff
@@ -15,8 +15,7 @@ def scan(report, logs)
       def ansible_report?(logs)
         return false if logs.blank?
         logs.any? do |log|
-          log['log'].fetch('messages', {}).
-            fetch('message', '') =~ /"_ansible_parsed"/
+          log['log'].fetch('reporter', nil) == 'ansible'
```
Comment (adamruzicka) [inline app/services/foreman_ansible/ansible_report_scanner.rb]: Isn't `a_hash.fetch('reporter', nil)` essentialy equivalent to `a_hash['reporter']`?

Comment (xprazak2) [reply app/services/foreman_ansible/ansible_report_scanner.rb]: Yeah, updated.

**Comment by adamruzicka** [comment]
Looks it should work with the linked ansible changes, will need to test once the ansible changes land.

**Comment by ares** [comment]
I raised a question in ansible PR, I think it would be better to add origin as top-level attribute than have it as part of each and every log

**Comment by xprazak2** [comment]
It would be, but report scanners do not have access to the top-level report attributes.

**Comment by sbernhard** [comment]
Any update on this PR?

**Comment by ezr-ondrej** [comment]
> Any update on this PR?

The ansible part still didn't make it in.

**Comment by xprazak2** [comment]
Now needs https://github.com/theforeman/foreman-ansible-modules/pull/896 and https://github.com/theforeman/foreman/pull/7819

File: webpack/components/ReportJsonViewer.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const theme = {
 
 const ReportJsonViewer = ({ data }) => (
   <div className="report-json-viewer">
-    <JSONTree data={this.props.data} hideRoot theme={theme} />
+    <JSONTree data={data} hideRoot theme={theme} />
```
Comment (ezr-ondrej) [inline webpack/components/ReportJsonViewer.js:13]: is this fix related?

File: app/services/foreman_ansible/ansible_report_scanner.rb [hunk 3]
```diff
@@ -5,19 +5,14 @@ module ForemanAnsible
...
     class << self
-      def scan(report, logs)
-        if (is_ansible = ansible_report?(logs))
-          report.origin = 'Ansible'
-        end
-        is_ansible
+      def add_reporter_data(report, raw); end
+
+      def identify_origin(raw)
+        raw['reporter']
```
Comment (ezr-ondrej) [inline app/services/foreman_ansible/ansible_report_scanner.rb:11]: This could accidently return something else, in case another config mgmt tool would use this field, right?
Following seems bit safer:
```suggestion
        'Ansible' if ansible_report?(raw)
```

File: webpack/components/ReportJsonViewer.js (hunk 2 above)
Comment (xprazak2) [reply webpack/components/ReportJsonViewer.js:13]: Yes, report is not displayed properly without it:

data-without

data-within

File: app/services/foreman_ansible/ansible_report_scanner.rb (hunk 3 above)
Comment (xprazak2) [reply app/services/foreman_ansible/ansible_report_scanner.rb]: Updated.

File: webpack/components/ReportJsonViewer.js (hunk 2 above)
Comment (ezr-ondrej) [reply webpack/components/ReportJsonViewer.js:13]: Thanks for explanation :+1:

**Comment by ezr-ondrej** [comment]
I've fixed the fixture for this, hopefully the tests are going to pass.

File: app/services/foreman_ansible/ansible_report_scanner.rb [hunk 4]
```diff
@@ -5,19 +5,14 @@ module ForemanAnsible
...
       end
 
-      def ansible_report?(logs)
-        return false if logs.blank?
-        logs.any? do |log|
-          log['log'].fetch('messages', {}).
-            fetch('message', '') =~ /"_ansible_parsed"/
-        end
+      def ansible_report?(raw)
+        raw['reporter'] == 'ansible'
```
Comment (ezr-ondrej) [inline app/services/foreman_ansible/ansible_report_scanner.rb:15]: Should we keep the backward compatibility?
By renaming the old method to `ansible_27_report?` and
```suggestion
        raw['reporter'] == 'ansible' || ansible_27_report?(raw['logs'] || [])
```

Comment (ezr-ondrej) [reply app/services/foreman_ansible/ansible_report_scanner.rb:15]: And we should then have two fixtures and two tests for it I'd guess :thinking:

**Comment by xprazak2** [comment]
I kept the old version of method as well.
